util/file_reader.js

```diff
@@ -0,0 +1,33 @@
+import fs from "fs";
+import path from "path";
+
+const MAX_DEPTH = 2;
+
+export function collectAllFileSources(fileOrDir, ext = null, depth = 0) {
```

Comment:
Is the `ext` argument necessary here? AFAICT, it's only being called with ".js" from the crawler. I could see it being useful to be able to specify e.g. `.ts` and `.js` extensions in the future, but possibly that could just be done within the function without needing an argument?